system/system_monitor/src/hdd_monitor/hdd_monitor.cpp

```diff
@@ -348,5 +335,119 @@ std::string HDDMonitor::getDeviceFromMountPoint(const std::string & mount_point)
   return ret;
 }
 
+bool HDDMonitor::updateHDDInfoList(HDDSMARTInfoItem item, diagnostic_updater::DiagnosticStatusWrapper & stat)
+{
+  static uint32_t size = static_cast<uint32_t>(HDDSMARTInfoItem::SIZE);
+  uint32_t item_index = static_cast<uint32_t>(item);
+  if (item_index >= size) {
+    stat.summary(DiagStatus::ERROR, "invalid S.M.A.R.T. information");
+    return false;
+  }
+
+  // Connect to HDD reader if the item is out of date
+  if (!is_up_to_date_[item_index]) {
+    if (!getHDDInfoListFromHDDReader(stat)) {
```

Comment:
@kk-inoue-esol Please consider to explicitly connect to HDD reader only one time without any condition before updating diagnostics status. e.g. Move connection handling to `update` method if no bugs, or move to another thread.

Reply:
I moved connection handling in ROS timer event. (`update` function is never called)